src/waypoint.jsx

```diff
@@ -333,13 +359,30 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.children) {
+      const child = React.Children.only(this.props.children);
+      if (!isDomElement(child)) {
```

Comment:
I wonder if we should move this check into `componentWillMount` and `componentWillReceiveProps`. Not sure if throwing an error in render will cause weird issues with React. Also, using CWRP will allow us to only do this check when `children` has changed.